app/lib/staypuft/network_query.rb

```diff
@@ -59,6 +59,23 @@ def subnet_for_host(subnet_type_name, host=@host)
       interface_hash_for_host(subnet_type_name, host)[:subnet]
     end
 
+    def interfaces_for_tenant_subnet(host=@host)
+      subnet = subnet_for_host(Staypuft::SubnetType::TENANT, host)
+      top_level_interface = host.interfaces.where(identifier:
+        interface_hash_for_host(Staypuft::SubnetType::TENANT,
+          host)[:interface]).first
```

Comment:
subnet_for_host calls interface_hash_for_host internally, so you're making this twice. Perhaps you should do this instead:
interface_hash = interface_hash_for_host(Staypuft::SubnetType::TENANT, host)
subnet = interface_hash[:subnet]
top_level_interface = host.interfaces.where(identifier: interface_hash[:interface]).first